hype-submitter/src/main/java/com/spotify/hype/Submitter.java

```diff
@@ -59,36 +64,71 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(Submitter.class);
 
-  private static final String GCS_STAGING_PREFIX = "spotify-hype-staging";
+  private static final String STAGING_PREFIX = "spotify-hype-staging";
 
   private final ClasspathInspector classpathInspector;
-  private final String bucketName;
+  private final URI stagingLocation;
 
   private final VolumeRepository volumeRepository;
   private final DockerRunner runner;
 
-  public static Submitter create(String bucketName, ContainerEngineCluster cluster) {
+  public static Submitter createLocal() throws IOException {
+    return Submitter.createLocal(DockerCluster.dockerCluster());
+  }
+
+  public static Submitter createLocal(final DockerCluster cluster) throws IOException {
+    Path stagingLocation = new File(System.getProperty("user.home")).toPath()
+        .resolve(".tmp")
+        .resolve(STAGING_PREFIX);
+    LOG.info("Local staging location is " + stagingLocation);
+    Files.createDirectories(stagingLocation);
+    final ClasspathInspector classpathInspector = forLoader(Submitter.class.getClassLoader());
+    return new Submitter(classpathInspector, stagingLocation.toString(), cluster);
+  }
+
+  public static Submitter create(String stagingLocation, ContainerEngineCluster cluster) {
     final ClasspathInspector classpathInspector = forLoader(Submitter.class.getClassLoader());
-    return create(classpathInspector, bucketName, cluster);
+    return create(classpathInspector, stagingLocation, cluster);
   }
 
-  public static Submitter create(
-      ClasspathInspector classpathInspector,
-      String bucketName, ContainerEngineCluster cluster) {
-    return new Submitter(classpathInspector, bucketName, cluster);
+  public static Submitter create(ClasspathInspector classpathInspector,
+                                 String stagingLocation,
+                                 ContainerEngineCluster cluster) {
+    return new Submitter(classpathInspector, stagingLocation, cluster);
   }
 
-  private Submitter(
-      ClasspathInspector classpathInspector, String bucketName,
-      ContainerEngineCluster cluster) {
-    this.bucketName = Objects.requireNonNull(bucketName);
+  private URI getStagingURI(String stagingLocation) {
+    checkNotNull(stagingLocation);
+    URI uri = URI.create(stagingLocation);
+    if (!uri.isAbsolute()) {
+      return URI.create("file://" + uri.toString());
+    } else {
+      return uri;
+    }
+  }
+
+  private Submitter(ClasspathInspector classpathInspector,
+                    String stagingLocation,
+                    ContainerEngineCluster cluster) {
+    this.stagingLocation = getStagingURI(stagingLocation);
```

Comment:
This should probably do a check on the stagingLocation to ensure it's a `gs:` uri. Otherwise we might run into thing being staged locally and the kube pod not being able to find the files.